### PR TITLE
Py3k

### PR DESCRIPTION
**Pull request by ocefpaf**
@ChrisBarker-NOAA I can squash the commits if you want, but I left them separated to make it easier to review.  Note that `from .cell_tree2d import CellTree` will change the way you run `py.test`.

### CODE REVIEW

**Comment by ocefpaf** [comment]
I just rebased to accommodate the  latest changes, but do not merge this yet.  I am adding some Travis-CI template too.

Can you enable Travis in this repo?

**Comment by ocefpaf** [comment]
All done.   Travis is working on my branch: https://travis-ci.org/ocefpaf/cell_tree2d/builds/96451780

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    - wget http://bit.ly/miniconda -O miniconda.sh
+    - bash miniconda.sh -b -p $HOME/miniconda
+    - export PATH="$HOME/miniconda/bin:$PATH"
+    - conda update --yes conda
+    - travis_retry conda create --yes -n TEST $CONDA --file requirements.txt
+    - source activate TEST
+    - travis_retry conda install --yes pytest
+
+script:
+    - python setup.py build_ext -i
```
Comment (ChrisBarker-NOAA) [inline .travis.yml]: interesting -- you don't want to do a full install here?

File: MANIFEST.in [hunk 2]
```diff
@@ -0,0 +1,5 @@
+include README.md
```
Comment (ChrisBarker-NOAA) [inline MANIFEST.in]: Is this so we get a proper sdist?

Comment (ocefpaf) [inline MANIFEST.in]: Yes.  (And that is installable with `pip install sdist.ext` )

File: .travis.yml (hunk 1 above)
Comment (ocefpaf) [inline .travis.yml]: `python setup.py test` will do that for us.  We can actually remove this line.  Let me do that and rebase.

File: setup.py [hunk 3]
```diff
@@ -2,14 +2,25 @@
...
 except ImportError:
-    print "You need setuptools to build this module"
+    print("You need setuptools to build this module.")
 
 from Cython.Build import cythonize
 
 import numpy as np #for the include dirs...
 import os, sys
 
+class PyTest(TestCommand):
```
Comment (ChrisBarker-NOAA) [inline setup.py]: nice! this is the place to put it.

File: requirements.txt [hunk 4]
```diff
@@ -0,0 +1,3 @@
+numpy
```
Comment (ChrisBarker-NOAA) [inline requirements.txt]: do we need this?

It's my undestanding that for a single pacakge, you put the requrements in the setup.py

requirements.txt is for applications, or systems, or ... stuff that is not a single package.

the idea is that pip install this_package will be able to resolve the dependencies.

Comment (ocefpaf) [inline requirements.txt]: I forgot this part in `setup.py`:

```python
with open('requirements.txt') as f:
    require = f.readlines()
install_requires = [r.strip() for r in require]
...
install_requires=install_requires,
...

```

I like the `requirements.txt`, but I have no strong feelings about it.  Adding the requirements to the `setup.py` file is also fine.  Let me know which way you want to go and I will fix the PR.

Comment (ChrisBarker-NOAA) [inline requirements.txt]: no -- this is good. as long as it's not in two places....

File: setup.py [hunk 5]
```diff
@@ -2,14 +2,25 @@
...
+        self.verbose = True
+
+    def run_tests(self):
+        import pytest
+        errno = pytest.main(self.test_args)
+        sys.exit(errno)
+
 
 include_dirs = [np.get_include(),
                 os.path.join('.', 'src')]
```
Comment (ChrisBarker-NOAA) [inline setup.py]: we have this:
```
if sys.platform.startswith('win'):
    # need the stdint header for Windows (VS2008)
    include_dirs.append(os.path.join('.','src', 'win_headers'))
```
which should only be necessary for py2.7 (old Visual Studio). We're not set up to test on Windows with py3.4 or 3.5 -- so I'm wary of editing this. But it shoudl probably by something like:

sys.platform.startswith('win') and sys.version_info.major <= 2:
    include_dirs.append(os.path.join('.','src', 'win_headers'))

File: requirements.txt (hunk 4 above)
Comment (ocefpaf) [inline requirements.txt]: Done!

File: setup.py (hunk 5 above)
Comment (ocefpaf) [inline setup.py]: I changed it to:

```python
sys.platform.startswith('win') and sys.version_info.major <= 2:
    include_dirs.append(os.path.join('.', 'src', 'win_headers'))
```

**Comment by ChrisBarker-NOAA** [comment]
awsome! thanks for the contribution Filipe, and thanks Jay for merging.

I'm confused about the best way to maintain the conda recipe. I think we have three for this one now: one in this repo, one in IOOS conda-recipes, and one in orr-conda-recipes.

Filipe, any ideas about how to keep this all sorted out and in sync??

**Comment by ocefpaf** [comment]
> Filipe, any ideas about how to keep this all sorted out and in sync?

@ChrisBarker-NOAA I sent you an e-mail about this.  We need to address the recipe forking issue.  Right now my short-term solution decide a canonical place for the recipes and use a script to get the recipe from that place (see https://github.com/ioos/conda-recipes/pull/554).

That approach has many advantages beyond joining efforts to maintain recipes in one place:
- brings the recipe from a canonical place to any another repository
- freedom to select a specific commit
- maintainers can override it anytime and replace it with another own recipe if needed (slow response/disagreement with the upstream recipe source)

In a smaller community this has worked very well.  We just implemented this in the IOOS/SciTools channels.  Not sure how that would scale to larger communities though.

**Pull request merged**